apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/Brevoppretter.kt

```diff
@@ -29,6 +29,49 @@ class Brevoppretter(
     private val db: BrevRepository,
     private val innholdTilRedigerbartBrevHenter: InnholdTilRedigerbartBrevHenter,
 ) {
+    suspend fun opprettBrevSomHarInnhold(
```

Comment:
Her er det sikkert noko eg ikkje ser eller skjønner, for meg så ser det ut som du no innfører nye (lange) metodar både her og i innhaldshentaren som gjer veldig mykje det samme som det som allereie fins?

Reply:
Ja, tanken her er å få delt opp mye av den tunge logikken som ligger under `opprettbrev` da mye som er der og "påkreves" ikke brukes av alle men samtidig holder alle som bruker den gissel for å ta stilling til det og det er ekstremt tungvint bruk synes jeg.
Det er nok mer vi kan fjerne i de to flytene enn jeg har gjort nå, men jeg vil gjerne begynne litt forsiktig da vi har 0 integrasjonstester her.
Feks brevparametere som skulle ta stilling til noe helt irrelevant som  `BrevDataRedigerbarRequest` bare for å tilfredstille den metoden for alle brevene som var definert der.
Så slipper vi også å definere en `BrevDataRedigerbarRequest` i `hentInndata` for noen av flytene som gjør det mye tydeligere hva de trenger.

Og på toppen av det hele slipper vi sende ned alle disse magiske lambda wrapperene. Men det er jo min personlige mening :D

Reply:
To protestar først:

1. Korvidt brevparametre definerer opp ein `BrevDataRedigerbarRequest` som `{ brevdata }` eller sender inn `brevdata` direkte som parameter ser eg vel ikkje den heilt store forskjellen på.

2. Høgare-ordens funksjonar er ikkje magiske, det er jo plain programmering og eit etablert og vanleg mønster.

Men er einig i at `innholdTilRedigerbartBrevHenter.hentInnData` er massiv, og at det er fint å få delt opp den. Og at det er fint å få forenkla for dei plassane i koden kor vi faktisk veit kva slags brevdata vi har. Kanskje kan vi klare oss med ein tynnare variant av `GenerellBrevData` også i dei høva.

Reply:
2. Kanskje ikke magiske men jeg synes ingenting om at det sendes x antall nivåer nedover i callstacken, men det setter et krav om at du må bruke ` BrevDataRedigerbarRequest` selvom du ikke trenger det.

Reply:
Du kan jo fint ta bort BrevDataRedigerbarRequest frå BrevParametre, og ta bort frå BrevRoute. Men det er vel detaljar, eg er einig i poenget om at der vi veit kva slags brevdata vi har, bør vi kunne sende inn det direkte, og så sleppe å forholde oss til mapping og request-objekt. Men veldig usikker på dupliseringsstrategien, det kjens sårbart for endringar. Utan at eg veit eller har skjønt heilt vegen vidare etter denne PR-en.

Ideelt sett kunne eg tenkt meg at `Brevoppretter.opprettBrev` tok inn brevdata som ein uniontype av `Brevdata | brevDataMapping: suspend (BrevDataRedigerbarRequest) -> BrevDataRedigerbar`. Skal tygge litt på om det fins ein lur kotlin-måte å gjera dét på.

Reply:
Greit nok at det er "plain programmering", men at noe _kan_ gjøres betyr ikke nødvendigvis at det _bør_ gjøres. Koden er forferdelig tung å lese når det sendes rundt ørten "HO-funksjoner" gjennom flere ulike nivåer og man må hoppe frem og tilbake mellom de ulike klassene for å få et mentalt bilde over hva som skjer hvor og hvorfor. 🤷‍♂️

Reply:
Heller også mot å gjøre det så enkelt som mulig, selv om dette kanskje vil medføre noe duplisering her 👍 Kjør på